File: android/src/oldarch/com/appodeal/rnappodeal/RNAppodealModule.java
package com.appodeal.rnappodeal;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;

public class RNAppodealModule extends ReactContextBaseJavaModule {

    private final RNAppodealModuleImpl moduleImplementation;

    public RNAppodealModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.moduleImplementation = new RNAppodealModuleImpl(reactContext);
    }

    @NonNull
    @Override
    public String getName() {
        return RNAppodealModuleImpl.NAME;
    }

    @ReactMethod
    public void initialize(String appKey, int adTypes, String pluginVersion) {
        moduleImplementation.initialize(appKey, (double) adTypes, pluginVersion);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public boolean isInitialized(int adTypes) {
        return moduleImplementation.isInitialized((double) adTypes);
    }

    @ReactMethod
    public void show(int adTypes, String placement) {
        moduleImplementation.show((double) adTypes, placement);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public boolean isLoaded(int adTypes) {
        return moduleImplementation.isLoaded((double) adTypes);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public boolean canShow(int adTypes, String placement) {
        return moduleImplementation.canShow((double) adTypes, placement);
    }

    @ReactMethod
    public void hide(int adTypes) {
        moduleImplementation.hide((double) adTypes);
    }

    @ReactMethod
    public void cache(int adTypes) {
        moduleImplementation.cache((double) adTypes);
    }

    @ReactMethod
    public void setAutoCache(int adTypes, boolean isEnabled) {
        moduleImplementation.setAutoCache((double) adTypes, isEnabled);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public boolean isPrecache(int adType) {
        return moduleImplementation.isPrecache((double) adType);
    }

    @ReactMethod
    public void setTabletBanners(boolean flag) {
        moduleImplementation.setTabletBanners(flag);
    }

    @ReactMethod
    public void setSmartBanners(boolean flag) {
        moduleImplementation.setSmartBanners(flag);
    }

    @ReactMethod
    public void setBannerAnimation(boolean flag) {
        moduleImplementation.setBannerAnimation(flag);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public int consentStatus() {
        return (int) moduleImplementation.consentStatus();
    }

    @ReactMethod
    public void requestConsentInfoUpdateWithAppKey(String appKey, final Promise promise) {
        moduleImplementation.requestConsentInfoUpdateWithAppKey(appKey, promise);
    }

    @ReactMethod
    public void showConsentFormIfNeeded(final Promise promise) {
        moduleImplementation.showConsentFormIfNeeded(promise);
    }

    @ReactMethod
    public void showConsentForm(final Promise promise) {
        moduleImplementation.showConsentForm(promise);
    }

    @ReactMethod
    public void revokeConsent() {
        moduleImplementation.revokeConsent();
    }

    @ReactMethod
    public void setChildDirectedTreatment(boolean flag) {
        moduleImplementation.setChildDirectedTreatment(flag);
    }

    @ReactMethod
    public void setTesting(boolean flag) {
        moduleImplementation.setTesting(flag);
    }

    @ReactMethod
    public void setLogLevel(String level) {
        moduleImplementation.setLogLevel(level);
    }

    @ReactMethod
    public void setTriggerPrecacheCallbacks(int adTypes, boolean flag) {
        moduleImplementation.setTriggerPrecacheCallbacks((double) adTypes, flag);
    }

    @ReactMethod
    public void disableNetwork(String networkName, int adTypes) {
        moduleImplementation.disableNetwork(networkName, (double) adTypes);
    }

    @ReactMethod
    public void setUserId(String id) {
        moduleImplementation.setUserId(id);
    }

    @ReactMethod
    public void setExtrasStringValue(String key, String value) {
        moduleImplementation.setExtrasStringValue(key, value);
    }

    @ReactMethod
    public void setExtrasDoubleValue(String key, double value) {
        moduleImplementation.setExtrasDoubleValue(key, value);
    }

    @ReactMethod
    public void setExtrasIntegerValue(String key, int value) {
        moduleImplementation.setExtrasIntegerValue(key, (double) value);
    }

    @ReactMethod
    public void setExtrasBooleanValue(String key, boolean value) {
        moduleImplementation.setExtrasBooleanValue(key, value);
    }

    @ReactMethod
    public void setExtrasMapValue(String key, ReadableMap value) {
        moduleImplementation.setExtrasMapValue(key, value);
    }

    @ReactMethod
    public void removeExtrasValue(String key) {
        moduleImplementation.removeExtrasValue(key);
    }

    @ReactMethod
    public void setCustomStateStringValue(String key, String value) {
        moduleImplementation.setCustomStateStringValue(key, value);
    }

    @ReactMethod
    public void setCustomStateDoubleValue(String key, double value) {
        moduleImplementation.setCustomStateDoubleValue(key, value);
    }

    @ReactMethod
    public void setCustomStateIntegerValue(String key, int value) {
        moduleImplementation.setCustomStateIntegerValue(key, (double) value);
    }

    @ReactMethod
    public void setCustomStateBooleanValue(String key, boolean value) {
        moduleImplementation.setCustomStateBooleanValue(key, value);
    }

    @ReactMethod
    public void setCustomStateMapValue(String key, ReadableMap value) {
        moduleImplementation.setCustomStateMapValue(key, value);
    }

    @ReactMethod
    public void removeCustomStateValue(String key) {
        moduleImplementation.removeCustomStateValue(key);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public WritableMap getRewardParameters(String placement) {
        return moduleImplementation.getRewardParameters(placement);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public double predictedEcpm(int adType) {
        return moduleImplementation.predictedEcpm((double) adType);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public String getPlatformSdkVersion() {
        return moduleImplementation.getPlatformSdkVersion();
    }

    @ReactMethod
    public void trackInAppPurchase(double amount, String currency) {
        moduleImplementation.trackInAppPurchase(amount, currency);
    }

    @ReactMethod
    public void validateAndTrackInAppPurchase(ReadableMap params, Promise promise) {
        moduleImplementation.validateAndTrackInAppPurchase(params, promise);
    }

    @ReactMethod
    public void trackEvent(String name, ReadableMap parameters) {
        moduleImplementation.trackEvent(name, parameters);
    }

    @ReactMethod
    public void setBidonEndpoint(String endpoint) {
        moduleImplementation.setBidonEndpoint(endpoint);
    }

    @ReactMethod(isBlockingSynchronousMethod = true)
    public String getBidonEndpoint() {
        return moduleImplementation.getBidonEndpoint();
    }

    @ReactMethod
    public void eventsNotifyReady(boolean ready) {
        moduleImplementation.eventsNotifyReady(ready);
    }

    @ReactMethod
    public void eventsAddListener(String eventName) {
        moduleImplementation.eventsAddListener(eventName);
    }

    @ReactMethod
    public void eventsRemoveListener(String eventName, boolean all) {
        moduleImplementation.eventsRemoveListener(eventName, all);
    }

    @ReactMethod
    public void eventsGetListeners(Promise promise) {
        moduleImplementation.eventsGetListeners(promise);
    }
}

File: android/src/newarch/com/appodeal/rnappodeal/RNAppodealModule.kt
package com.appodeal.rnappodeal

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.WritableMap
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = RNAppodealModuleImpl.NAME)
class RNAppodealModule(
    private val reactContext: ReactApplicationContext
) : NativeAppodealModuleSpec(reactContext) {

    private val moduleImplementation by lazy { RNAppodealModuleImpl(reactContext) }

    override fun getName(): String = RNAppodealModuleImpl.NAME

    // MARK: - NativeAppodealSpec Implementation

    override fun initialize(appKey: String, adTypes: Double, pluginVersion: String) {
        moduleImplementation.initialize(appKey, adTypes, pluginVersion)
    }

    override fun isInitialized(adTypes: Double): Boolean {
        return moduleImplementation.isInitialized(adTypes)
    }

    override fun show(adTypes: Double, placement: String) {
        moduleImplementation.show(adTypes, placement)
    }

    override fun isLoaded(adTypes: Double): Boolean {
        return moduleImplementation.isLoaded(adTypes)
    }

    override fun canShow(adTypes: Double, placement: String): Boolean {
        return moduleImplementation.canShow(adTypes, placement)
    }

    override fun hide(adTypes: Double) {
        moduleImplementation.hide(adTypes)
    }

    override fun cache(adTypes: Double) {
        moduleImplementation.cache(adTypes)
    }

    override fun setAutoCache(adTypes: Double, value: Boolean) {
        moduleImplementation.setAutoCache(adTypes, value)
    }

    override fun isPrecache(adTypes: Double): Boolean {
        return moduleImplementation.isPrecache(adTypes)
    }

    override fun setTabletBanners(value: Boolean) {
        moduleImplementation.setTabletBanners(value)
    }

    override fun setSmartBanners(value: Boolean) {
        moduleImplementation.setSmartBanners(value)
    }

    override fun setBannerAnimation(value: Boolean) {
        moduleImplementation.setBannerAnimation(value)
    }

    override fun consentStatus(): Double {
        return moduleImplementation.consentStatus()
    }

    override fun revokeConsent() {
        moduleImplementation.revokeConsent()
    }

    override fun requestConsentInfoUpdateWithAppKey(appKey: String, promise: Promise) {
        moduleImplementation.requestConsentInfoUpdateWithAppKey(appKey, promise)
    }

    override fun showConsentFormIfNeeded(promise: Promise) {
        moduleImplementation.showConsentFormIfNeeded(promise)
    }

    override fun showConsentForm(promise: Promise) {
        moduleImplementation.showConsentForm(promise)
    }

    override fun setChildDirectedTreatment(value: Boolean) {
        moduleImplementation.setChildDirectedTreatment(value)
    }

    override fun setTesting(value: Boolean) {
        moduleImplementation.setTesting(value)
    }

    override fun setLogLevel(value: String) {
        moduleImplementation.setLogLevel(value)
    }

    override fun setTriggerPrecacheCallbacks(adTypes: Double, value: Boolean) {
        moduleImplementation.setTriggerPrecacheCallbacks(adTypes, value)
    }

    override fun disableNetwork(network: String, adTypes: Double) {
        moduleImplementation.disableNetwork(network, adTypes)
    }

    override fun getPlatformSdkVersion(): String {
        return moduleImplementation.getPlatformSdkVersion()
    }

    override fun setUserId(id: String) {
        moduleImplementation.setUserId(id)
    }

    override fun setExtrasStringValue(key: String, value: String) {
        moduleImplementation.setExtrasStringValue(key, value)
    }

    override fun setExtrasIntegerValue(key: String, value: Double) {
        moduleImplementation.setExtrasIntegerValue(key, value)
    }

    override fun setExtrasDoubleValue(key: String, value: Double) {
        moduleImplementation.setExtrasDoubleValue(key, value)
    }

    override fun setExtrasBooleanValue(key: String, value: Boolean) {
        moduleImplementation.setExtrasBooleanValue(key, value)
    }

    override fun setExtrasMapValue(key: String, value: ReadableMap) {
        moduleImplementation.setExtrasMapValue(key, value)
    }

    override fun removeExtrasValue(key: String) {
        moduleImplementation.removeExtrasValue(key)
    }

    override fun setCustomStateStringValue(key: String, value: String) {
        moduleImplementation.setCustomStateStringValue(key, value)
    }

    override fun setCustomStateIntegerValue(key: String, value: Double) {
        moduleImplementation.setCustomStateIntegerValue(key, value)
    }

    override fun setCustomStateDoubleValue(key: String, value: Double) {
        moduleImplementation.setCustomStateDoubleValue(key, value)
    }

    override fun setCustomStateBooleanValue(key: String, value: Boolean) {
        moduleImplementation.setCustomStateBooleanValue(key, value)
    }

    override fun setCustomStateMapValue(key: String, value: ReadableMap) {
        moduleImplementation.setCustomStateMapValue(key, value)
    }

    override fun removeCustomStateValue(key: String) {
        moduleImplementation.removeCustomStateValue(key)
    }

    override fun getRewardParameters(placement: String): WritableMap {
        return moduleImplementation.getRewardParameters(placement)
    }

    override fun predictedEcpm(adType: Double): Double {
        return moduleImplementation.predictedEcpm(adType)
    }

    override fun trackInAppPurchase(amount: Double, currency: String) {
        moduleImplementation.trackInAppPurchase(amount, currency)
    }

    override fun setBidonEndpoint(endpoint: String) {
        moduleImplementation.setBidonEndpoint(endpoint)
    }

    override fun getBidonEndpoint(): String? {
        return moduleImplementation.getBidonEndpoint()
    }

    override fun validateAndTrackInAppPurchase(purchase: ReadableMap, promise: Promise) {
        moduleImplementation.validateAndTrackInAppPurchase(purchase, promise)
    }

    override fun trackEvent(name: String, parameters: ReadableMap) {
        moduleImplementation.trackEvent(name, parameters)
    }

    override fun eventsNotifyReady(ready: Boolean) {
        moduleImplementation.eventsNotifyReady(ready)
    }

    override fun eventsAddListener(eventName: String) {
        moduleImplementation.eventsAddListener(eventName)
    }

    override fun eventsRemoveListener(eventName: String, all: Boolean) {
        moduleImplementation.eventsRemoveListener(eventName, all)
    }

    override fun eventsGetListeners(promise: Promise) {
        moduleImplementation.eventsGetListeners(promise)
    }
}
